src/main/webapp/app/exercises/programming/manage/services/programming-exercise-simulation.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { SERVER_API_URL } from 'app/app.constants';
+import { EntityResponseType, ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable({ providedIn: 'root' })
+export class ProgrammingExerciseSimulationService {
+    public resourceUrl = SERVER_API_URL + 'api/programming-exercises';
+
+    constructor(private http: HttpClient, private programmingExerciseService: ProgrammingExerciseService) {}
+
+    /**
+     * Triggers the creation and setup of a new programming exercise
+     * @param programmingExercise
+     */
+    automaticSetupWithoutLocalSetup(programmingExercise: ProgrammingExercise): Observable<EntityResponseType> {
```

Comment:
maybe we should integrate some kind of guard here that fails automatically during production

Reply:
I added guards for the methods, which should be not used in production